### PR TITLE
(fix) Continue the architecture work

### PR DESCRIPTION
**Pull request by nilscognite**
#### Type of change

Feat  

#### Jira ticket :blue_book:

https://cognitedata.atlassian.net/browse/

## Description :pencil:

* Fix issues that @haakonflatval-cognite addressed (removed utility functions, classes, unclear code in `DomainObjectChange`, and commented `roundIncrement()`)
* Add comments around
* Some renaming to make it even more consistent
* Add some utility function in `DomainObject` that was missing
* Moved some functions around
* Fixed some minor user experience bugs during creation of measurements
* Fixed some corner cases in the Axis visualization.
* Added support for crop box, but on in UI yet. 

## Checklist :ballot_box_with_check:

- [x] I am proud of this feature.
- [x] have performed a self-review of my own code.
- [x] I have added PR type (Feat, Bug, Chore, Test, Docs, Style, Refactor) to the PR title.
- [ ] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation.
- [ ] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [ ] I have checked that my changes do not introduce regressions in the public documentation.
- [ ] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [ ] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [ ] I have added TSDoc to any public facing changes.
- [ ] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [ ] I have noted down and am currently tracking any technical debt introduced in this PR.
- [ ] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: react-components/src/architecture/concrete/axis/AxisThreeView.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export class AxisThreeView extends GroupThreeView {
 
   private readonly _corners: Vector3[];
   private readonly _faceCenters: Vector3[];
-  private readonly _sceneBoundingBox: Box3 = new Box3().makeEmpty(); // Caching the bounding box of the scene
+  private readonly _sceneBoundingBox: Box3 = new Box3().makeEmpty(); // Casehing the bounding box of the scene
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/concrete/axis/AxisThreeView.ts:45]: :monocle_face:

File: react-components/src/architecture/concrete/axis/AxisThreeView.ts [hunk 2]
```diff
@@ -607,6 +601,48 @@ function getTickDirection(faceIndex1: number, faceIndex2: number, target: Vector
   return target;
 }
 
+function incrementToString(value: number): string {
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/concrete/axis/AxisThreeView.ts:604]: Could we instead use this function? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Number/toPrecision

Comment (nilscognite) [reply react-components/src/architecture/concrete/axis/AxisThreeView.ts:604]: Jeg slet med den funksjonen. Nei, vi kan ikke bruke slike fuksjoner. Vi vet vi ikke noe om presisjonen eller mangnituden av value. Kan vøre hva som helst egentlig.

File: react-components/src/architecture/concrete/axis/AxisThreeView.ts (hunk 1 above)
Comment (nilscognite) [reply react-components/src/architecture/concrete/axis/AxisThreeView.ts]: Fixed

File: react-components/src/architecture/concrete/axis/AxisThreeView.ts [hunk 3]
```diff
@@ -42,7 +42,7 @@ export class AxisThreeView extends GroupThreeView {
 
   private readonly _corners: Vector3[];
   private readonly _faceCenters: Vector3[];
-  private readonly _sceneBoundingBox: Box3 = new Box3().makeEmpty(); // Caching the bounding box of the scene
+  private readonly _sceneBoundingBox: Box3 = new Box3().makeEmpty(); // Cashe the bounding box of the scene
```
Comment (haakonflatval-cognite) [inline react-components/src/architecture/concrete/axis/AxisThreeView.ts:45]: :monocle_face:

**Review by haakonflatval-cognite — APPROVED** [review]
LGTM :+1:

**Pull request merged**